ibm/service/usagereports/data_source_ibm_billing_snapshot_list.go

```diff
@@ -0,0 +1,413 @@
+							Computed:    true,
+							Description: "Version of the snapshot.",
+						},
+						"created_on": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Date and time of creation of snapshot.",
+						},
+						"report_types": {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "List of report types configured for the snapshot.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The type of billing report of the snapshot. Possible values are [account_summary, enterprise_summary, account_resource_instance_usage].",
+									},
+									"version": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "Version of the snapshot.",
+									},
+								},
+							},
+						},
+						"files": {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "List of location of reports.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"report_types": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "The type of billing report stored. Possible values are [account_summary, enterprise_summary, account_resource_instance_usage].",
+									},
+									"location": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "Absolute path of the billing report in the COS instance.",
+									},
+									"account_id": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "Account ID for which billing report is captured.",
+									},
+								},
+							},
+						},
+						"processed_at": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Timestamp at which snapshot is captured.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIBMBillingSnapshotListRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	usageReportsClient, err := meta.(conns.ClientSession).UsageReportsV4()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getReportsSnapshotOptions := &usagereportsv4.GetReportsSnapshotOptions{}
+
+	userDetails, err := meta.(conns.ClientSession).BluemixUserDetails()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getReportsSnapshotOptions.SetAccountID(userDetails.UserAccount)
+	getReportsSnapshotOptions.SetMonth(d.Get("month").(string))
+	if _, ok := d.GetOk("date_from"); ok {
+		getReportsSnapshotOptions.SetDateFrom(int64(d.Get("date_from").(int)))
+	}
+	if _, ok := d.GetOk("date_to"); ok {
+		getReportsSnapshotOptions.SetDateTo(int64(d.Get("date_to").(int)))
+	}
+
+	snapshotList, response, err := usageReportsClient.GetReportsSnapshotWithContext(context, getReportsSnapshotOptions)
```

Comment:
Pagination is not handled in this datasource..
Kindly handle pagination and remove first and last arguments from schema.

After removing these arguments from schema remove from docs as well.

Reply:
Added pagination support.

Reply:
Removed the first and next arguments from the schema as well.